kpops/components/streams_bootstrap_v3/base.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import annotations
+
+import logging
+import re
+from abc import ABC
+from typing import TYPE_CHECKING, Any
+
+import pydantic
+from pydantic import AliasChoices, ConfigDict, Field
+
+from kpops.component_handlers.helm_wrapper.model import HelmRepoConfig
+from kpops.components.base_components import KafkaApp
+from kpops.components.base_components.helm_app import HelmApp, HelmAppValues
+from kpops.components.common.topic import KafkaTopic, KafkaTopicStr
+from kpops.utils.docstring import describe_attr
+from kpops.utils.pydantic import (
+    CamelCaseConfigModel,
+    DescConfigModel,
+    exclude_by_value,
+    exclude_defaults,
+)
+
+if TYPE_CHECKING:
+    try:
+        from typing import Self  # pyright: ignore[reportAttributeAccessIssue]
+    except ImportError:
+        from typing_extensions import Self
+
+
+STREAMS_BOOTSTRAP_HELM_REPO = HelmRepoConfig(
+    repository_name="bakdata-streams-bootstrap",
+    url="https://bakdata.github.io/streams-bootstrap/",
+)
+STREAMS_BOOTSTRAP_VERSION = "3.0.0"
+STREAMS_BOOTSTRAP_VERSION_PATTERN = r"^(\d+)\.(\d+)\.(\d+)(-[a-zA-Z]+(\.[a-zA-Z]+)?)?$"
+COMPILED_VERSION_PATTERN = re.compile(STREAMS_BOOTSTRAP_VERSION_PATTERN)
+
+# Source of the pattern: https://kubernetes.io/docs/concepts/containers/images/#image-names
+IMAGE_TAG_PATTERN = r"^[a-zA-Z0-9_][a-zA-Z0-9._-]{0,127}$"
+
+log = logging.getLogger("StreamsBootstrapV3")
+
+
+class StreamsBootstrapV3Values(HelmAppValues):
+    """Base value class for all streams bootstrap related components.
+
+    :param image_tag: Docker image tag of the streams-bootstrap app.
+    :param kafka: Kafka configuration for the streams-bootstrap app.
+    """
+
+    image_tag: str = Field(
+        default="latest",
+        pattern=IMAGE_TAG_PATTERN,
+        description=describe_attr("image_tag", __doc__),
+    )
+
+    kafka: KafkaConfig = Field(
+        default=...,
+        description=describe_attr("kafka", __doc__),
+    )
+
+
+class StreamsBootstrapV3(KafkaApp, HelmApp, ABC):
+    """Base for components with a streams-bootstrap Helm chart.
+
+    :param values: streams-bootstrap Helm values
+    :param repo_config: Configuration of the Helm chart repo to be used for
+        deploying the component, defaults to streams-bootstrap Helm repo
+    :param version: Helm chart version, defaults to "3.0.0"
+    """
+
+    values: StreamsBootstrapV3Values = Field(
+        default_factory=StreamsBootstrapV3Values,
+        description=describe_attr("values", __doc__),
+    )
+
+    repo_config: HelmRepoConfig = Field(
+        default=STREAMS_BOOTSTRAP_HELM_REPO,
+        description=describe_attr("repo_config", __doc__),
+    )
+
+    version: str = Field(
+        default=STREAMS_BOOTSTRAP_VERSION,
+        pattern=STREAMS_BOOTSTRAP_VERSION_PATTERN,
+        description=describe_attr("version", __doc__),
+    )
+
+    @pydantic.model_validator(mode="after")
```

Comment:
shouldn't it be a field validator?
otherwise, it could also be builtin to the pattern where we only allow 3 as the major version

Reply:
I couldn't make it work with the field validator. The pattern didn't work either. :/

Reply:
update: made it work 😉